src/irc_bridge.py

```diff
@@ -32,44 +32,49 @@ def __init__(self, message: str):
         super().__init__(message)
 
 
-class IRCBridge(ops.Object):  # pylint: disable=too-few-public-methods
+class IRCBridgeObserver(ops.Object):  # pylint: disable=too-few-public-methods
     """A class representing the IRC bridge plugin for Synapse application.
 
     See https://github.com/matrix-org/matrix-appservice-irc/ for more details about it.
     """
 
-    def __init__(self, charm: ops.CharmBase, charm_state: CharmState):
+    def __init__(self, charm: CharmBaseWithState) -> None:
         """Initialize a new instance of the IRC bridge class.
 
         Args:
             charm: The charm object that the IRC bridge instance belongs to.
-            charm_state: Instance of CharmState.
         """
         super().__init__(charm, "irc-bridge")
         self._charm = charm
-        self._charm_state = charm_state
         self.framework.observe(charm.on.collect_unit_status, self._on_collect_status)
 
-    @property
-    def _pebble_service(self) -> typing.Any:
-        """Return instance of pebble service.
+    def get_charm(self) -> CharmBaseWithState:
+        # pylint:disable=duplicate-code
+        # pylint is extra picky over the smallest details
+        # this doesn't need refactoring
+        """Return the current charm.
 
         Returns:
-            instance of pebble service or none.
+            The charm object.
         """
-        return getattr(self._charm, "pebble_service", None)
+        return self._charm
 
-    def _on_collect_status(self, event: ops.CollectStatusEvent) -> None:
+    @inject_charm_state
```

Comment:
Oh ok I think I remember this, I think the guidance is changed to not use a decorator and instead construct it in the method? @gregory-schiano perhaps a PR to take a look at

Reply:
Well the implementation on Synapse crossed the finalisation of our decision. For coherence purpose I'd keep it this way, and we'll tackle that in a separate track when we'll ensure the new guidance is implemented on all our charms